catalyst/contrib/dl/callbacks/periodical_loader.py

```diff
@@ -0,0 +1,103 @@
+from typing import Mapping
+from collections import OrderedDict
+import copy
+
+from torch.utils.data import DataLoader
+
+from catalyst.core import Callback, CallbackOrder, State
+
+
+class PeriodicLoaderRunnerCallback(Callback):
+    """Callback for runing loaders with specified period.
+    To disable loader use ``0`` as period.
+
+    .. note::
+        ``'train'`` is a required loader and will
+        be ignored from passed loaders.
+
+    Example:
+
+        >>> PeriodicLoaderRunnerCallback(
+        >>>     train_additional=2,
+        >>>     valid=3,
+        >>>     valid_additional=5
+        >>> )
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Args:
+            kwargs: loader names and their run periods.
+
+                .. warning::
+                    Loder will be ignored if a period
+                    value is not a number or loader
+                    name is not presented in a list of loaders.
+        """
+        super().__init__(order=CallbackOrder.External)
+
+        self.valid_loader: str = None
+        self.valid_metrics: Mapping[str, float] = None
+        self.loaders: Mapping[str, DataLoader] = OrderedDict()
+
+        required_callbacks = {"train"}
+        self.loader_periods = {}
+        for loader, period in kwargs.items():
+            if not isinstance(loader, str) or not isinstance(
+                period, (int, float)
+            ):
```

Comment:
what do you think about assert here, to make sure, that all passed params are correct? 
I think,it's better to raise an error, rather then silently "forget" about some loader :)

Reply:
What do you think about raising `TypeError` with some message instead of assert statement?

Reply:
`TypeError` looks good :+1: